service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ApplicantControllerTest.java

```diff
@@ -0,0 +1,150 @@
+
+        // then
+        assertThat(callbackResponse.getErrors()).doesNotContain(ERROR_MESSAGE);
+
+        String actualPbaNumber = extractPbaNumberFromApplicant(callbackResponse);
+        assertThat(actualPbaNumber).isEqualTo("PBA1234567");
+    }
+
+    @Test
+    void shouldReturnNoErrorsWhenPbaNumberStartsWithPBAAndIsSevenDigits() throws Exception {
+        // given
+        String pbaNumber = "PBA1234567";
+
+        // when
+        AboutToStartOrSubmitCallbackResponse callbackResponse = makeRequest(createApplicant(pbaNumber));
+
+        // then
+        assertThat(callbackResponse.getErrors()).doesNotContain(ERROR_MESSAGE);
+        String actualPbaNumber = extractPbaNumberFromApplicant(callbackResponse);
+        assertThat(actualPbaNumber).isEqualTo("PBA1234567");
+    }
+
+    @Test
+    void shouldReturnNoErrorsWhenPbaNumberIsNotEntered() throws Exception {
+        // given
+        String pbaNumber = null;
+
+        // when
+        AboutToStartOrSubmitCallbackResponse callbackResponse = makeRequest(createApplicant(pbaNumber));
+
+        // then
+        assertThat(callbackResponse.getErrors()).doesNotContain(ERROR_MESSAGE);
+        String actualPbaNumber = extractPbaNumberFromApplicant(callbackResponse);
+        assertThat(actualPbaNumber).isNull();
+    }
+
+    @SuppressWarnings("unchecked")
+    private String extractPbaNumberFromApplicant(AboutToStartOrSubmitCallbackResponse callbackResponse) {
+        Map<String, Object> map = callbackResponse.getData();
+        Map<String, Object> applicantMap = (Map<String, Object>) map.get("applicant");
+        String actualPbaNumber = (String) applicantMap.get("pbaNumber");
+        return actualPbaNumber;
+    }
+
+    private Applicant createApplicant(String pbaNumber) {
+        return new Applicant("","", "", "", null,
+            "", "", "", pbaNumber);
+    }
+
+    private AboutToStartOrSubmitCallbackResponse makeRequest(Applicant applicant) throws Exception {
+        HashMap<String, Object> map = MAPPER.readValue(MAPPER.writeValueAsString(applicant),
+            new TypeReference<Map<String, Object>>() {
+            });
+
+        CallbackRequest request = CallbackRequest.builder()
+            .caseDetails(CaseDetails.builder()
+                .id(12345L)
+                .data(ImmutableMap.<String, Object>builder().put("applicant", map).build())
+                .build())
+            .build();
+
+        MvcResult response = mockMvc
+            .perform(post("/callback/enter-applicant/mid-event")
+                .header("authorization", AUTH_TOKEN)
+                .header("user-id", USER_ID)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(MAPPER.writeValueAsString(request)))
+            .andExpect(status().isOk())
+            .andReturn();
+
+        return MAPPER.readValue(response.getResponse()
+            .getContentAsByteArray(), AboutToStartOrSubmitCallbackResponse.class);
+    }
+}
+
+
```

Comment:
Remove white space, we keep one line space at the end of files only. I'm surprised the linter did not flag.

Reply:
Done